### PR TITLE
Add process() error handling and add usage

### PR DESCRIPTION
**Pull request by plutov**
I found that CLI usage is not clear when you don't pass in, out options:
```
triangle -in ~/Downloads/GO_BUILD.png ~/Downloads/GO_BUILD2.png
Generating triangulated image... /Saved as: . ✓
```
or:
```
triangle
2017/12/22 15:47:32 stat : no such file or directory
```

So in this PR I changed it to show user that these options are mandatory:
```
go run cmd/triangle/main.go
2017/12/22 15:55:33 usage: triangle -in inout.jpg -out out.jpg
```

Also I found that in one place `process()` error is not handled, so I added Print for it.

And I executed `gofmt` on this file, so it changed indents a bit.

### CODE REVIEW

File: cmd/triangle/main.go [hunk 1]
```diff
@@ -58,9 +59,13 @@ func main() {
 	var wg sync.WaitGroup
 	flag.Parse()
 
+	if len(*source) == 0 || len(*destination) == 0 {
```
Comment (plutov) [inline cmd/triangle/main.go:62]: Added this.

File: cmd/triangle/main.go [hunk 2]
```diff
@@ -141,7 +146,7 @@ func main() {
 		// Drain the channel.
 		for f := range ch {
 			if f.err != nil {
-				fmt.Printf("Error converting image: %s ", f.img.Name())
+				fmt.Printf("Error converting image: %s: %s", f.img.Name(), err.Error())
```
Comment (plutov) [inline cmd/triangle/main.go:149]: Added error, not only file name.

File: cmd/triangle/main.go [hunk 3]
```diff
@@ -151,9 +156,13 @@ func main() {
 		if err != nil {
 			log.Fatal(err)
 		}
-		process(file, *destination)
-		fmt.Printf("Saved as: %s \x1b[92m✓\n\n", path.Base(*destination))
-		
+		f, processErr := process(file, *destination)
```
Comment (plutov) [inline cmd/triangle/main.go:159]: Added error handling.

**Review by esimov — APPROVED** [review]
Thanks for PR. `gofmt` sometimes does not format the code on my style, but i will take care about this lately.

**Pull request merged**